File: supabase/seeds/_import.mjs
import { createClient } from '../../node_modules/@supabase/supabase-js/dist/module/index.js';
import { readFileSync, readdirSync } from 'fs';
import { join, dirname } from 'path';
import { fileURLToPath } from 'url';

// Parse .env.local manually
const envPath = join(dirname(fileURLToPath(import.meta.url)), '../../.env.local');
const env = Object.fromEntries(
  readFileSync(envPath, 'utf8')
    .split('\n')
    .filter(l => l.includes('=') && !l.startsWith('#'))
    .map(l => { const i = l.indexOf('='); return [l.slice(0,i).trim(), l.slice(i+1).trim()]; })
);

const supabase = createClient(env.NEXT_PUBLIC_SUPABASE_URL, env.SUPABASE_SERVICE_ROLE_KEY);

const seedDir = dirname(fileURLToPath(import.meta.url));
const files = readdirSync(seedDir).filter(f => f.endsWith('.sql') && !f.startsWith('_')).sort();

for (const file of files) {
  const sql = readFileSync(join(seedDir, file), 'utf8');

  // Extract values between the $JSON$ delimiters
  const jsonMatch = sql.match(/\$JSON\$([\s\S]*?)\$JSON\$/);
  if (!jsonMatch) { console.error(`No JSON found in ${file}`); continue; }

  // Extract section_slug, page_slug, title from the VALUES block
  const slugsMatch = sql.match(/VALUES\s*\(\s*'([^']+)',\s*'([^']+)',\s*'([^']+)'/);
  if (!slugsMatch) { console.error(`Could not parse slugs in ${file}`); continue; }

  const [, section_slug, page_slug, title] = slugsMatch;
  let content;
  try {
    content = JSON.parse(jsonMatch[1].trim());
  } catch (e) {
    console.error(`Invalid JSON in ${file}:`, e.message);
    continue;
  }

  const { error } = await supabase.from('documents').upsert(
    { section_slug, page_slug, title, content, updated_by: 'system' },
    { onConflict: 'section_slug,page_slug' }
  );

  if (error) {
    console.error(`❌ ${file}: ${error.message}`);
  } else {
    console.log(`✅ ${section_slug}/${page_slug} — ${title}`);
  }
}

File: supabase/seeds/_import.py
#!/usr/bin/env python3
import os, re, json, urllib.request, urllib.error
from pathlib import Path

# Parse .env.local
env = {}
env_path = Path(__file__).parent.parent.parent / '.env.local'
for line in env_path.read_text().splitlines():
    if '=' in line and not line.startswith('#'):
        k, _, v = line.partition('=')
        env[k.strip()] = v.strip()

url_base = env['NEXT_PUBLIC_SUPABASE_URL'].rstrip('/')
service_key = env['SUPABASE_SERVICE_ROLE_KEY']
endpoint = f"{url_base}/rest/v1/documents"

headers = {
    'Authorization': f'Bearer {service_key}',
    'apikey': service_key,
    'Content-Type': 'application/json',
    'Prefer': 'resolution=merge-duplicates,return=minimal',
}

seeds_dir = Path(__file__).parent
sql_files = sorted(f for f in seeds_dir.iterdir() if f.suffix == '.sql' and not f.name.startswith('_'))

for sql_file in sql_files:
    sql = sql_file.read_text()

    json_match = re.search(r'\$JSON\$([\s\S]*?)\$JSON\$', sql)
    slugs_match = re.search(r"VALUES\s*\(\s*'([^']+)',\s*'([^']+)',\s*'([^']+)'", sql)

    if not json_match or not slugs_match:
        print(f"❌ Could not parse {sql_file.name}")
        continue

    section_slug, page_slug, title = slugs_match.groups()
    try:
        content = json.loads(json_match.group(1).strip())
    except json.JSONDecodeError as e:
        print(f"❌ Invalid JSON in {sql_file.name}: {e}")
        continue

    payload = json.dumps({
        'section_slug': section_slug,
        'page_slug': page_slug,
        'title': title,
        'content': content,
        'updated_by': 'system',
    }).encode()

    req = urllib.request.Request(endpoint, data=payload, headers=headers, method='POST')
    try:
        with urllib.request.urlopen(req) as resp:
            print(f"✅ {section_slug}/{page_slug} — {title}")
    except urllib.error.HTTPError as e:
        body = e.read().decode()
        print(f"❌ {section_slug}/{page_slug}: HTTP {e.code} — {body[:200]}")
